node/packages/parser/src/ethereum/yearn.ts

```diff
@@ -0,0 +1,33 @@
+import { Tx } from '@shapeshiftoss/blockbook'
+import { TxSpecific as ParseTxSpecific, YearnTx } from '../types'
+import shapeShiftRouter from './abi/shapeShiftRouter'
+import yearnVault from './abi/yearnVault'
+import { addABI, decodeMethod } from 'abi-decoder'
+import { GenericParser } from './index'
+
+const ROUTER_CONTRACT = '0x6a1e73f12018D8e5f966ce794aa2921941feB17E'
+
+export class Parser implements GenericParser {
+  supportsTransaction(tx: Tx) {
+    const data = tx.ethereumSpecific?.data
+    const firstVoutAddress = tx.vout && tx.vout[0].addresses && tx.vout[0].addresses[0]
+    const contractData = data?.slice(0, 2).concat(data?.slice(34, 74)).toLowerCase()
```

Comment:
This looks like `contractAddress` vs `contractData` and is this supposed to be checking for the address paramter against multiple contract functions that happen to be in the same position?

I think I would rather be explicit in our conditions and verify specific arguments in calling functions than some magic position of an address in the input data.

Reply:
I've fixed this up in https://github.com/shapeshift/unchained/pull/267/commits/2953d65c9a978276a520a2f9707dcb3c1381583e by using the `args._spender` on the decoded transaction. I don't like that the `_` indicates this probably shouldn't be directly accessed, but I'm not sure how else to do it.

Reply:
The `_` doesn't have the same private meanings as far as solidity is concerned. Typically just used to indicate a function parameter in the case of

```
address contract;

someFunc(address _contract) public {
    contract = _contract;
}
```